File: internal/service/user_service.go
package service

import "log"

type UserService struct {
	logger *log.Logger
}

func New(logger *log.Logger) *UserService {
	return &UserService{
		logger: logger,
	}
}

func (a UserService) Register() {

}

func (a UserService) GetUserByIdx() {

}

File: server/user-service/src/main/java/org/wid/userservice/service/UserService.java
package org.wid.userservice.service;

import org.springframework.stereotype.Service;
import org.wid.userservice.dto.user.UserDto;
import org.wid.userservice.entity.entity.User;
import org.wid.userservice.mapper.UserMapper;
import org.wid.userservice.port.primary.UserServicePort;
import org.wid.userservice.port.secondary.UserRepositoryPort;

import lombok.RequiredArgsConstructor;
import reactor.core.publisher.Mono;

@Service
@RequiredArgsConstructor
public class UserService implements UserServicePort {

  private final UserRepositoryPort userRepository;
  private final UserMapper userMapper;

  @Override
  public Mono<UserDto> getUser(String userId) {
    Mono<User> user = userRepository.getUserById(userId);

    return user.map(userMapper::entityToUserDto);
  }
}
